src/rust/guya/template/src/template.rs

```diff
@@ -0,0 +1,251 @@
+		.json()
+		.expect("Manga detail json not found")
+		.as_object()?;
+	let title = json
+		.get("title")
+		.as_string()
+		.expect("Manga detail title not found")
+		.read();
+	let cover = format!(
+		"{}{}",
+		&data.base_url,
+		json.get("cover").as_string()?.read()
+	);
+	let description_raw = json.get("description").as_string()?.read();
+	let description_node = Node::new_fragment(description_raw.as_bytes())?;
+	let description = match description_node.select("body").array().get(0).as_node() {
+		Ok(node) => node.own_text().read(),
+		Err(_) => String::from(""),
+	};
+	let user_url = format!("{}/read/manga/{}/", &data.base_url, slug);
+	let author = match json.get("author").as_string() {
+		Ok(author) => author.read(),
+		Err(_) => String::from("Unknown Author"),
+	};
+	let artist = match json.get("artist").as_string() {
+		Ok(artist) => artist.read(),
+		Err(_) => String::from("Unknown Artist"),
+	};
+	Ok(Manga {
+		id: slug,
+		title,
+		cover,
+		description,
+		author,
+		artist,
+		url: user_url,
+		status: MangaStatus::Unknown,
+		nsfw,
+		viewer: MangaViewer::Rtl,
+		..Default::default()
+	})
+}
+
+pub fn get_chapter_list(data: GuyaSiteData, slug: String) -> Result<Vec<Chapter>> {
+	let url = format!("{}/api/series/{}/", &data.base_url, slug);
+	let request = Request::new(url, HttpMethod::Get).header("User-Agent", "Aidoku");
+	let json = request
+		.json()
+		.expect("Manga chapter list json not found")
+		.as_object()?;
+	let mut chapter_arr: Vec<Chapter> = Vec::new();
+	let chapter_obj = json.get("chapters").as_object()?;
+	let mut chapters: Vec<f32> = chapter_obj
+		.keys()
+		.map(|k| k.as_string().unwrap().read().parse::<f32>().unwrap())
+		.collect();
+	chapters.sort_by(|a, b| b.partial_cmp(a).unwrap());
+	for chapter in chapters {
+		let chapter = chapter.to_string();
+		let obj = match chapter_obj.get(&chapter).as_object() {
+			Ok(obj) => obj,
+			Err(_) => continue,
+		};
+		let title = obj.get("title").as_string()?.read();
+		let volume = obj
+			.get("volume")
+			.as_string()?
+			.read()
+			.parse()
+			.unwrap_or(-1.0);
+		let chapter_int = chapter.parse().unwrap_or(1.0);
+		let user_url = format!("{}/read/manga/{}/{}/", &data.base_url, &slug, chapter);
+		let folder = obj.get("folder").as_string()?.read();
+		for groups in obj.get("groups").as_object()?.keys() {
+			let group_id = groups.as_string()?.read();
+			let mut group_name = String::new();
+			let group_list = json.get("groups").as_object()?;
+			for gl_index in group_list.keys() {
+				let gl_id = gl_index.as_string()?.read();
+				if gl_id == group_id {
+					group_name = group_list.get(&gl_id).as_string()?.read();
+				}
+			}
+			let mut date_updated = 0.0;
+			let date_list = obj.get("release_date").as_object()?;
+			for dl_index in date_list.keys() {
+				let dl_id = dl_index.as_string()?.read();
+				if dl_id == group_id {
+					date_updated = date_list.get(&dl_id).as_float()?;
+				}
+			}
+			let title = &title;
+			let url = &user_url;
```

Comment:
why do you do this...? just do `title.clone()` in the chapter struct constructor because that's what you're doing anyways

Reply:
LMAOO I HAVE NO CLUE EITHER